src/audio/module_adapter/module/dts.c

```diff
@@ -334,16 +346,16 @@ static int dts_codec_apply_config(struct processing_module *mod)
 		if (param->size > param_header_size) {
 			/* Calculate size of param->data */
 			param_data_size = param->size - param_header_size;
+			comp_info(dev, "dts_codec_apply_config() id %d size %d",
+					param->id, param_data_size);
 
 			if (param_data_size) {
 				dts_result = dtsSofInterfaceApplyConfig(codec->private, param->id,
 					param->data, param_data_size);
-				ret = dts_effect_convert_sof_interface_result(dev, dts_result);
-				if (ret) {
+				if (dts_result != DTS_SOF_INTERFACE_RESULT_SUCCESS) {
 					comp_err(dev,
 						"dts_codec_apply_config() dtsSofInterfaceApplyConfig failed %d",
 						dts_result);
-					return ret;
```

Comment:
@joechengxperi how is this change related to the addition of the free memory callback?